controllers/controllers.go

```diff
@@ -84,6 +119,51 @@ func (r *instanceReconciler) ensureInstanceIsUpToDate(instanceInfo *instance.Inf
 	return nc.Configure()
 }
 
+// upgradeBlocked returns whether the upgrade should be blocked if an upgrade will break storage functionality.
+func (r *instanceReconciler) upgradeBlocked(node *core.Node, upgradingToCSI bool) bool {
+	// Only consider the case of vSphere and Azure. Safely migrating in-tree storage on other platforms is not supported
+	// by WMCO.
+	if r.platform != config.VSpherePlatformType && r.platform != config.AzurePlatformType {
+		return false
+	}
+	if !upgradingToCSI {
+		return false
+	}
+	if _, present := node.GetLabels()[AllowUpgradeLabel]; present {
+		return false
+	}
+	if value := node.GetLabels()[CSIConfiguredLabel]; value == "true" {
+		return false
+	}
+	if len(node.Status.VolumesAttached) == 0 {
```

Comment:
Looking after `VolumesInUse` may be more accurate, given `VolumesAttached` includes unused volumes that are not attached to any pods.

Reply:
I was advised by the storage team to use VolumesAttached